packages/main/src/plugin/authentication.ts

```diff
@@ -0,0 +1,189 @@
+    const onDidChangeSessionDisposable = provider.onDidChangeSessions(
+      (event: AuthenticationProviderAuthenticationSessionsChangeEvent) => {
+        this._onDidChangeSessions.fire({ provider: { id, label } });
+        this.apiSender.send('authentication-provider-update');
+      },
+    );
+    return {
+      dispose: () => {
+        onDidChangeSessionDisposable.dispose();
+        this._authenticationProviders.delete(id);
+      },
+    };
+  }
+
+  addAccountUsage(providerId: string, accountLabel: string, extensionId: string, extensionName: string): void {
+    throw new Error('The method is not implemented!');
+  }
+
+  readAllowedExtensions(providerId: string, accountName: string): AllowedExtension[] {
+    throw new Error('The method is not implemented!');
+  }
+
+  updateAllowedExtension(
+    providerId: string,
+    accountName: string,
+    extensionId: string,
+    extensionName: string,
+    isAllowed: boolean,
+  ): void {
+    throw new Error('The method is not implemented!');
+  }
+
+  isAccessAllowed(providerId: string, accountName: string, extensionId: string): boolean | undefined {
+    return true; // To be implemented later
+  }
+
+  async getSession(
+    requestingExtension: ExtensionInfo,
+    providerId: string,
+    scopes: string[],
+    options: AuthenticationGetSessionOptions & { createIfNone: true },
+  ): Promise<AuthenticationSession | undefined>;
+  async getSession(
+    requestingExtension: ExtensionInfo,
+    providerId: string,
+    scopes: string[],
+    options?: AuthenticationGetSessionOptions,
+  ): Promise<AuthenticationSession | undefined>;
+  async getSession(
+    requestingExtension: ExtensionInfo,
+    providerId: string,
+    scopes: string[],
+    options: AuthenticationGetSessionOptions = {},
+  ): Promise<AuthenticationSession | undefined> {
+    // Error cases
+    if (options.forceNewSession) {
+      throw new Error('Option is not supported. Please remove forceNewSession option.');
+    }
+    if (options.silent) {
+      throw new Error('Option is not supported. Please remove silent option.');
+    }
+    if (options.clearSessionPreference) {
+      throw new Error('Option is not supported. Please remove clearSessionPreference option.');
+    }
+
+    const provider = this._authenticationProviders.get(providerId)?.provider;
+    if (!provider) {
+      // window.showInformationMessage(`Requested authentication provider ${providerId} is not installed.`);
+    } else {
+      const sessions = await provider.getSessions(scopes);
+      if (sessions.length > 0 && this.isAccessAllowed(providerId, sessions[0].account.label, requestingExtension.id)) {
+        return sessions[0];
+      } else {
+        return provider.createSession(scopes);
+      }
+    }
+    return;
+  }
+
+  async removeSession(providerId: string, sessionId: string): Promise<void> {
+    const provider = this._authenticationProviders.get(providerId)?.provider;
+    return provider?.removeSession(sessionId);
```

Comment:
```
 * If the removal is successful, the onDidChangeSessions event should be fired.
```
we're missing the event being fired

```
     * If a session cannot be removed, the provider should reject with an error message.
```

not finding the provider id is probably an error ?

Reply:
provider will fire the event when session is removed and it is going not be propagated to UI. All other issues should be fixed.